torch/csrc/Module.cpp

```diff
@@ -1010,6 +1010,7 @@ Call this whenever a new thread is created in order to propagate values from
 
   ASSERT_TRUE(set_module_attr("has_cuda", has_cuda));
   ASSERT_TRUE(set_module_attr("has_mps", has_mps));
+  ASSERT_TRUE(set_module_attr("is_mps_available", at::hasMPS() ? Py_True : Py_False));
```

Comment:
@albanD , is this something you had in mind ? the hasMPS() in the global context uses DeviceGuardImpl to get the information if MPS is loaded.

Reply:
That looks right yes.

Reply:
The issue was with the availability macros API used. I had to switch to `__builtin_available`. The function is consistent with other functions like `hasMKL` etc.